### ISSUE TITLE
Do not convert APIs not available in Swift

### ISSUE THREAD

**Issue by Ibrahimhass** [Swiftify-Corp/Swiftify#160]
Here is the Objective C code.

```
NSInvocationOperation * operation = [[NSInvocationOperation alloc] initWithTarget:self selector:@selector(invokeOpenDB) object:nil];
```

Converted Swift Code:

```
let operation = NSInvocationOperation(target: self, selector: #selector(invokeOpenDB), object: nil)
```

But **NSInvocation** is not available in [Swift](https://share.getcloudapp.com/E0u9mZv5).

**Comment by alex-swiftify** [comment]
@Ibrahimhass I know that NSInvocation isn't available in Swift, but can you suggest any straightforward change to the converter to cover this?
We have a backlog task to translate NSInvocation to `DispatchQueue.main.async` here:
https://swiftify.atlassian.net/browse/SWC-805
... but this looks way too difficult, and may not be worth the efforts at all.

If you find any drop-in Swift replacement, i.e. like the SwiftTryCatch library, we could use that.

**Comment by Ibrahimhass** [comment]
@alex-swiftify This is one GitHub Project which can be used.
https://github.com/piotrtobolski/SwiftyInvocation

Please let me know if this can be used, if so then I will update it to Swift 5 and we can use it in the same way as `SwiftTryCatch`

**Comment by alex-swiftify** [comment]
@Ibrahimhass Feel free to try updating the project to the latest Swift version.
If most changes would be changing `NSInvocation` to `SIInvocation`, or other involved classes with `NS` prefix to `SI`, that's fine.

After a very quick look, the library could help to port the code from the summary of https://swiftify.atlassian.net/browse/SWC-805,
but it doesn't contain a drop-in replacement for `NSInvocationOperation` used in this ticket.
See if you can implement a corresponding `SIInvocationOperation` class easily.

**Comment by alex-swiftify** [comment]
@Ibrahimhass If this is doable in under 2 hours you may update and publish that project.
Otherwise, we'll postpone this.

**Comment by Ibrahimhass** [comment]
@alex-swiftify I have updated the project to Swift 5 and can successfully get the output for the test case mentioned here
```
let array: NSArray = ["1", "2", "3"]

let selector = #selector(NSArray.object(at:))
let methodSignature = NSArray.si_instanceMethodSignature(for: selector)
let invocation = SIInvocation(methodSignature: methodSignature)
invocation.target = array
invocation.selector = selector
var argument = 1
invocation.setArgument(&argument, at: 2)
invocation.retainArguments()

invocation.invoke()

var returnValue: Unmanaged<NSString>?
invocation.getReturnValue(&returnValue)

XCTAssertEqual(returnValue?.takeUnretainedValue(), "2")
```

But for cases where the function needs a parameter or returns a value type, the program crashes as we have to use the 
swift_getImplementation function.

Somewhat related StackOverflow [answer](https://stackoverflow.com/a/43714950/).

This might take longer than 2 hours so lets postpone this for now.

**Comment by alex-swiftify** [comment]
@Ibrahimhass I got you. Let's either push the update to Swift 5.x as the PR to the original repo, or fork SwiftyInvocation in the Swiftify repo.

**Comment by Ibrahimhass** [comment]
@alex-swiftify I have a forked version on my GitHub profile where the issues were already fixed earlier.
https://github.com/Ibrahimhass/SwiftyInvocation

I have also forked a version of the original repo under Swiftify
https://github.com/Swiftify-Corp/SwiftyInvocation.

**Comment by alex-swiftify** [comment]
@Ibrahimhass Isn't it more logical to have the version updated to Swift 5 under Swiftify?

**Comment by Ibrahimhass** [comment]
@alex-swiftify Yes, without a doubt, I had earlier forked it to my profile [here](https://github.com/Ibrahimhass/SwiftyInvocation), the Swift 5 code had some issues I was thinking of fixing [this issue](https://take.ms/lwYjV) first and then update the repository [here](https://github.com/Swiftify-Corp/SwiftyInvocation)

**Comment by alex-swiftify** [comment]
@Ibrahimhass Well, if we agreed that we postpone this, maybe you just fork from your repository to Swiftify for now.
Don't spend a lot of time on this.

**Comment by Ibrahimhass** [comment]
@alex-swiftify I have forked my repo to Swiftify now.
https://github.com/Swiftify-Corp/SwiftyInvocation
